File: java-pr/49/java/performance/TextAnalyzer17.java
// {fact rule=performance@v1.0 defects=1}
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class TextAnalyzer17 {
    private Database database = new Database();
    
    public int countWords(String text) {
        String[] words = text.split(" ");
        int count = 0;
        for (String word : words) {
            if (!word.isEmpty()) {
                count++;
            }
        }
        return count;
    }
    
    private static class Database {
        public User findUserById(int id) { return new User(); }
        public List<User> findAllUsers() { return new ArrayList<>(); }
    }
    
    private static class User {}
}
// {/fact}

File: js-pr/56/javascript/code-quality-performance/text_analyzer_17.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function countWords(text) {
    let words = text.split(" ");
    let count = 0;
    for (let word of words) {
        if (word) {
            count++;
        }
    }
    return count;
}
// {/fact}